processor/resourcedetectionprocessor/internal/aws/ec2/ec2_test.go

```diff
@@ -0,0 +1,104 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ec2
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+type mockMetadata struct {
+	ret       ec2metadata.EC2InstanceIdentityDocument
+	returnErr error
+}
+
+func (mm mockMetadata) get(ctx context.Context) (ec2metadata.EC2InstanceIdentityDocument, error) {
+	if mm.returnErr != nil {
+		return ec2metadata.EC2InstanceIdentityDocument{}, mm.returnErr
+	}
+	return mm.ret, nil
+}
+
+func TestDetector_Detect(t *testing.T) {
+	type fields struct {
+		provider ec2MetadataProvider
+	}
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    pdata.Resource
+		wantErr bool
+	}{
+		{"success", fields{provider: &mockMetadata{ec2metadata.EC2InstanceIdentityDocument{
+			Region:           "us-west-2",
+			AccountID:        "account1234",
+			AvailabilityZone: "us-west-2a",
+			InstanceID:       "i-abcd1234",
+			ImageID:          "abcdef",
+			InstanceType:     "c4.xlarge",
+		}, nil}}, args{ctx: context.Background()}, func() pdata.Resource {
```

Comment:
nit: this is a little hard to read.  mind separating a bit and using uniform indentation?

Reply:
See if you think it's any better now.